web/src/app/validation/ceg/duplicate-linked-node.ts

```diff
@@ -0,0 +1,30 @@
+import { CEGLinkedNode } from 'src/app/model/CEGLinkedNode';
+import { CEGModel } from '../../model/CEGModel';
+import { IContainer } from '../../model/IContainer';
+import { Type } from '../../util/type';
+import { ElementValidatorBase } from '../element-validator-base';
+import { ValidationMessage } from '../validation-message';
+import { ValidationResult } from '../validation-result';
+import { Validator } from '../validator-decorator';
+
+type IO = ('input' | 'output');
+
+@Validator(CEGModel)
+export class DublicateLinkedNodeValidator extends ElementValidatorBase<CEGModel> {
+    public validate(element: CEGModel, contents: IContainer[]): ValidationResult {
+
+        const invalidNodes: CEGLinkedNode[] = [];
+        let linkedNodes: CEGLinkedNode[] = contents
+            .filter(element => Type.is(element, CEGLinkedNode))
+            .map(element => element as CEGLinkedNode);
+        for (const node of linkedNodes) {
+            if (linkedNodes.filter(element => element.linkTo.url === node.linkTo.url).length > 1) {
```

Comment:
Won't that mark all linked nodes? I was expecting a filter that `element !== node && element.linkTo.url === node.linkTo.url`.